backend/tests/modules/v2/test_allocations_v2.py

```diff
@@ -0,0 +1,462 @@
+from datetime import datetime
+import pytest
+from httpx import AsyncClient
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from tests.modules.v2.utils import FakeAllocationRequest, FakeUser, FakeAllocation
+
+
+class TestGetDonorsForEpochV1:
+    """Test cases for the GET /allocations/donors/{epoch_id} (get_donors_for_epoch_v1) endpoint"""
+
+    @pytest.mark.asyncio
+    async def test_returns_empty_list_when_no_donors(
+        self, fast_client: AsyncClient, fast_session: AsyncSession
+    ):
+        """Should return an empty list when there are no donors for the epoch"""
+
+        async with fast_client as client:
+            # When: requesting donors for epoch 1
+            resp = await client.get("allocations/donors/1")
+
+            # Then: response should be successful with empty donors list
+            assert resp.status_code == 200
+            assert resp.json() == {"donors": []}
+
+    @pytest.mark.asyncio
+    async def test_returns_donors_when_they_exist(
+        self, fast_client: AsyncClient, fast_session: AsyncSession
+    ):
+        """Should return a list of donors when they exist"""
+
+        # Given: a donor
+        alice = await FakeUser.GetAlice(fast_session)
+        await FakeAllocation.of_(fast_session, alice, 1)
+        await FakeAllocation.of_(fast_session, alice, 2)
+        await FakeAllocation.of_(fast_session, alice, 3)
+
+        bob = await FakeUser.GetBob(fast_session)
+        await FakeAllocation.of_(fast_session, bob, 1)
+        await FakeAllocation.of_(fast_session, bob, 2)
+
+        charlie = await FakeUser.GetCharlie(fast_session)
+        await FakeAllocation.of_(fast_session, charlie, 1)
+
+        await fast_session.commit()
```

Comment:
if we decide not to drop tables but work on a session, then we don't have to commit all changes separately because everything will be in the same session: https://github.com/golemfoundation/octant/pull/569/files#diff-3a776b82682c765eab3df0c5ca8321dbbfb0427bce808a05eb712f4ae6b902c3R51